providers/ms365/resources/ms365.lr

```diff
@@ -390,4 +393,8 @@ ms365.sharepointonline {
 ms365.teams {
   // CS Teams client configuration
   csTeamsClientConfiguration dict
+    // CS Teams Tenant Federated configuration
+  CsTenantFederationConfiguration dict
```

Comment:
a few things here: 
  * property names are lower case, `csTenantFederalConfiguration`
  * I am not sure what the `cs` here stands for, can we remove it? Looks like a prefix for their powershell cmdlets.
  * I would suggest we make a proper resource here instead of using a dict. Let's ensure new resources are properly typed so that we do not run into sudden issues if the API/cmdlets start returning different things. You could define the resource as following:
  ```
  ms365.teams.tenantFederationConfig {
  .....
  }
  ```
  and use all the things you need. We can start with just the properties you actually need and add things on a use-case basis

You can the reference in in ms365.teams as 
```
ms365.teams {
....
tenantFederationConfiguration ms365.teams.tenantFederationConfiguration
..
}
```